lib/rules/no-property-in-node.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const { ESLintUtils } = require('@typescript-eslint/utils');
+const tsutils = require('ts-api-utils');
+
+const typedNodeSourceFileTesters = [
+  /@types[/\\]estree[/\\]index\.d\.ts/,
+  /@typescript-eslint[/\\]types[/\\]dist[/\\]generated[/\\]ast-spec\.d\.ts/,
+];
+
+/**
+ * @param {import('typescript').Type} type
+ * @returns Whether the type seems to include a known ESTree or TSESTree AST node.
+ */
```

Comment:
I'm a bit unfamiliar with some of the utils and concepts here so maybe including an example in the comment for this function would help explain what it's doing?

Reply:
Does 3575644 answer what you're looking for?

Reply:
Thanks!